cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -144,11 +144,15 @@ public AbstractAppFabricHttpHandler(Authenticator authenticator) {
   }
 
   protected int getInstances(HttpRequest request) throws IOException, NumberFormatException {
-    return parseBody(request, Instances.class).getInstances();
+    Instances instances = parseBody(request, Instances.class);
```

Comment:
Instances can be null when `request.getContent()` is not readable. It is may not be due to user error. We can catch `NumberFormatException` or `JsonSyntaxException` in callers and send BAD_REQUEST. I will add that too.